std/shared/src/main/scala/collection/Vector.scala

```diff
@@ -0,0 +1,35 @@
+package scalaz
+package std
+
+import typeclass.{ EqClass, MonadClass }
+
+trait VectorInstances {
+  implicit val vectorMonad: Monad[Vector] = instanceOf(new MonadClass[Vector] {
+    override def map[A, B](fa: Vector[A])(f: A => B): Vector[B]             = fa.map(f)
+    override def ap[A, B](fa: Vector[A])(f: Vector[A => B]): Vector[B]      = fa.zip(f).map { case (e, f) => f(e) }
```

Comment:
for stdlib stuff I wonder if it makes sense to manually fuse operations like this:

```scala
(fa.iterator zip f.iterator).map({ case (e, f) => f(e) })(breakOut)
```

/me ducks